servers/physics_2d/body_2d_sw.cpp

```diff
@@ -384,11 +384,20 @@ void Body2DSW::set_space(Space2DSW *p_space) {
 
 void Body2DSW::_compute_area_gravity_and_dampenings(const Area2DSW *p_area) {
 	if (p_area->is_gravity_point()) {
-		if (p_area->get_gravity_distance_scale() > 0) {
-			Vector2 v = p_area->get_transform().xform(p_area->get_gravity_vector()) - get_transform().get_origin();
-			gravity += v.normalized() * (p_area->get_gravity() / Math::pow(v.length() * p_area->get_gravity_distance_scale() + 1, 2));
+		const real_t gravity_distance_scale = p_area->get_gravity_distance_scale();
+		Vector2 v = p_area->get_transform().xform(p_area->get_gravity_vector()) - get_transform().get_origin();
+		if (gravity_distance_scale > 0) {
+			real_t v_length = v.length();
+			if (v_length > 0) {
+				Vector2 new_gravity = gravity + (v.normalized() * (p_area->get_gravity() / Math::pow(v_length * gravity_distance_scale, 2)));
```

Comment:
`Math::pow` could be replaced with a simple multiplication here for performance reasons. Most compilers would optimize it anyway, but I'm not sure it's the case for all supported compilers and platforms so better be safe and use the simpler code.